core/src/main/java/edu/wpi/grip/core/events/StepAddedEvent.java

```diff
@@ -11,23 +15,23 @@
  */
 public class StepAddedEvent {
     private final Step step;
-    private final Optional<Integer> index;
+    private final OptionalInt index;
 
     /**
      * @param step  The step being added
      * @param index The index that the step should be added at.
      */
-    public StepAddedEvent(Step step, int index) {
-        this.step = step;
-        this.index = Optional.of(index);
+    public StepAddedEvent(Step step, @Nonnegative int index) {
```

Comment:
https://github.com/WPIRoboticsProjects/GRIP/pull/392#discussion_r51906483

Also, this class was using the wrong optional type.